### ISSUE TITLE
IE 10 Support: Bugs & Fixes

### ISSUE THREAD

**Issue by sam-chuang** [hyperapp/hyperapp#466]
[Line 177](https://github.com/hyperapp/hyperapp/blob/master/src/index.js#L177)
```javascript
element = parent.insertBefore(createElement(node, isSVG), element)
```

Refer to
[IE 10 Error: Invalid Argument on insertBefore](https://github.com/MithrilJS/mithril.js/issues/353)

**Comment by JorgeBucaran** [comment]
@sam-chuang Thanks for the report! Does this mean we need to check if `element` is undefined, and then use `appendChild` instead? 🤔

**Comment by sam-chuang** [comment]
Yes, I tried 

```
element = parent.insertBefore(createElement(node, isSVG), element || null);
```

And it solve the error on IE 10

**Comment by JorgeBucaran** [comment]
@sam-chuang Great! Thanks.

**Comment by JorgeBucaran** [comment]
@sam-chuang I found that we call `insertBefore` in 3 different places, so we actually need to check if the second argument is undefined and make sure to pass null in every case. I think I am going to drop official IE10 support, because I don't have the time or energy to test in IE10.

**Comment by sam-chuang** [comment]
Hi @JorgeBucaran 

Sorry for late response, after look again this issue, I think I found the root cause, refer to [line 32](https://github.com/hyperapp/hyperapp/blob/master/src/index.js#L32)

`var root = container && container.children[0]`

At first, container doesn't have any children, so it's undefined, so I plan to fix it by

`var root = (container && container.children[0]) || null`

then IE 10 works, if you are feel ok about the code change above, I can create a PR to fix it

**Comment by JorgeBucaran** [comment]
@sam-chuang Sounds good, open a PR, please. By the way, would the following also work?

```js
container.children[0] = null
app(state, view, actions, container)
```

**Comment by sam-chuang** [comment]
WOW! Your fix also works on IE 10

But I am confused, your fix since for by app developer who care about IE 10 support, does it means you prefer not to fix inside Hyperrapp src?

**Comment by JorgeBucaran** [comment]
@sam-chuang 🎉  

As you know, IE is tricky, maybe the fix you suggest only works in some situation, but maybe we'll find another problem in the future. I'll reopen this issue to track IE related bugs and their fixes here. Then, when we have collected more data we can make a more informed decision. 

TL;DR: Please use my fix and let me know if you find another IE10 problem.

**Comment by sam-chuang** [comment]
Got it, thanks~~

Then I think I need to drop the PR I just created #529

**Comment by JorgeBucaran** [comment]
This is in master! 🎉

**Comment by infinnie** [comment]
Just by the way: I used to have a version of HyperApp for supporting **IE 8**.

https://github.com/infinnie/infinnie.github.io/issues/12
